code/datums/spells/devil.dm

```diff
@@ -0,0 +1,242 @@
+		stop_pulling()
+	src.loc = holder
+	src.holder = holder
+	src.notransform = 0
+	fakefireextinguish()
+
+/mob/living/proc/infernalphasein()
+	if(src.notransform)
+		to_chat(src,"<span class='warning'>You're too busy to jaunt in.</span>")
+		return 0
+	fakefire()
+	src.loc = get_turf(src)
+	src.client.eye = src
+	src.visible_message("<span class='warning'><B>[src] appears in a firey blaze!</B>")
+	playsound(get_turf(src), 'sound/misc/exit_blood.ogg', 100, 1, -1)
+	spawn(15)
+		fakefireextinguish(TRUE)
+
+/obj/effect/proc_holder/spell/targeted/sintouch
+	name = "Sin Touch"
+	desc = "Subtly encourage someone to sin."
+	charge_max = 1800
+	clothes_req = 0
+	selection_type = "range"
+	range = 2
+	cooldown_min = 0
+	overlay = null
+	include_user = 0
+	action_icon_state = "sintouch"
+	action_background_icon_state = "bg_demon"
+	random_target = 1
+	random_target_priority = TARGET_RANDOM
+	max_targets = 3
+	invocation = "TASTE SIN AND INDULGE!!"
+	invocation_type = "shout"
+
+/obj/effect/proc_holder/spell/targeted/sintouch/ascended
+	name = "Greater sin touch"
+	charge_max = 100
+	range = 7
+	max_targets = 10
+
+/obj/effect/proc_holder/spell/targeted/sintouch/cast(list/targets, mob/living/user = usr)
+	for(var/mob/living/carbon/human/H in targets)
+		if(!H.mind)
+			continue
+		for(var/datum/objective/sintouched/A in H.mind.objectives)
+			continue
+		H.influenceSin()
+		H.Weaken(2)
+		H.Stun(2)
+
+/obj/effect/proc_holder/spell/targeted/summon_dancefloor
+	name = "Summon Dancefloor"
+	desc = "When what a Devil really needs is funk."
+	include_user = 1
+	range = -1
+	clothes_req = 0
+
+	school = "conjuration"
+	charge_max = 10
+	cooldown_min = 50 //5 seconds, so the smoke can't be spammed
+	action_icon_state = "funk"
+	action_background_icon_state = "bg_demon"
+
+	var/list/dancefloor_turfs
+	var/list/dancefloor_turfs_types
+	var/dancefloor_exists = FALSE
+//	var/datum/effect_system/smoke_spread/transparent/dancefloor_devil/smoke
+
+
+/obj/effect/proc_holder/spell/targeted/summon_dancefloor/cast(list/targets, mob/user = usr)
+	LAZYINITLIST(dancefloor_turfs)
+	LAZYINITLIST(dancefloor_turfs_types)
+
+/*
+	if(!smoke)
+		smoke = new()
+	smoke.set_up(0, get_turf(user))
+	smoke.start()
+*/
+
+	if(dancefloor_exists)
+		dancefloor_exists = FALSE
+		for(var/i in 1 to dancefloor_turfs.len)
+			var/turf/T = dancefloor_turfs[i]
+			T.ChangeTurf(dancefloor_turfs_types[i])
+	else
+		dancefloor_exists = TRUE
+		var/i = 1
+		var/list/funky_turfs = RANGE_TURFS(1, user)
+		dancefloor_turfs.len = funky_turfs.len
+		dancefloor_turfs_types.len = funky_turfs.len
+		for(var/t in funky_turfs)
+			var/turf/T = t
+			dancefloor_turfs[i] = T
+			dancefloor_turfs_types[i] = T.type
+			T.ChangeTurf((i % 2 == 0) ? /turf/simulated/floor/light/colour_cycle/dancefloor_a : /turf/simulated/floor/light/colour_cycle/dancefloor_b)
+			i++
+
+/*
+/datum/effect_system/smoke_spread/transparent/dancefloor_devil
+	effect_type = /obj/effect/particle_effect/smoke/transparent/dancefloor_devil
```

Comment:
Yeah, tg's got a bit more consistent of an effects system, classed appropriately - this is a subtype of the smoke spreading effect system, which we don't have the appropriate refactors for yet - and I don't want to staple a miscellaneous refactor onto this gamemode PR.